### DISCUSSION TITLE
triggerValidation doesn't change the state of the form

### DISCUSSION THREAD

**Discussion by r-tae** [react-hook-form/react-hook-form#2128] [Q&A] [answered]
(on v5 by the way)

I suspect I'm misinterpreting the documentation but I'm calling triggerValidation inside a useEffect. The promise it returns eventually returns true, but my form's `isValid` stays false. Does anyone know what I'm missing? Thanks!

**Comment by bluebill1049** [comment]
can you share your codesandbox?

**Reply by r-tae** [reply]
https://codesandbox.io/s/rhf-mui-select-73p3c?file=/src/index.js
I'm actually using `yup` but this has the same behaviour

**Reply by bluebill1049** [reply]
try to console log `isValid` at the render phase.

**Comment by r-tae** [comment, marked as answer]
Ended up clearing up when I moved from:
```jsx
<Controller
  as={
    <FormControlLabel control={<Radio />} />
  }
/>
```
to the actually correct:
```jsx
<FormControlLabel
  control={
    <Controller as={<Radio />} />
  }
/>
```

**Reply by bluebill1049** [reply]
👍
